File: Submarino.cpp
#include "Submarino.h"

Submarino::Submarino(string n, int c, double l, int a, int p, int t)
    : Embarcacion(n, c, l, a) {
    profundidadMax = p;
    tripulantes = t;
    cout << "Submarino creado: " << nombre << endl;
}

Submarino::~Submarino() {
    cout << "Submarino destruido: " << nombre << endl;
}

void Submarino::sumergir() {
    cout << nombre << " se sumerge hasta " << profundidadMax << " metros." << endl;
}

void Submarino::mostrarDatos() {
    cout << "\n=== DATOS DEL SUBMARINO ===" << endl;
    Embarcacion::mostrarDatos();
    cout << "Profundidad máxima: " << profundidadMax << " m" << endl;
    cout << "Tripulantes: " << tripulantes << endl;
}

File: Submarino.py
from embarcacion import Embarcacion

class Submarino(Embarcacion):
    def __init__(self, nombre, capacidad, longitud, profundidad_max):
        super().__init__(nombre, "Submarino", capacidad, longitud)
        self.profundidad_max = profundidad_max

    def mostrar_info(self):
        super().mostrar_info()
        print(f"Profundidad máxima: {self.profundidad_max} metros")

    def sumergir(self):
        print(f"El submarino {self.nombre} se sumerge hasta {self.profundidad_max} metros.")
